### ISSUE TITLE
Is it possible to imlpement angular authguard service into Express server middleware using decorator?

### ISSUE THREAD

**Issue by sheshtiwari** [serhiisol/decorators-server#1]
I am learning node decorator using typescript. may you please give some guidance on how I can achieve this using the DI.

Angular authguard implementation
@injectable() 
class AuthGuards implement canActivate {

  constructor(private configurationService: ConfigurationService, private authService: AuthService, 
     private requestService:  RequestService){
   }

    # Function from canactivate interface
   CanActivate(){
    # somelogic here
   }
}

I want to use it within Express middleware -
const app = express;
app.use((req, res, next) => {
@middleware()
#some authGurard instance here to check authentication and initialize configuration of app
})

Please Guide me. Thanks in advance.

**Comment by serhiisol** [comment]
Not really sure I understand your question. Do you want to use angular and decorators as a SSR project ?

**Comment by sheshtiwari** [comment]
Yes SSR for react project. A service which implement canActivate interface and have few dependencies in constructor. I want to instantiate in express middleware for authentication check like in angular. Service Dependencies have their own other dependencies.

Container.get(AuthGuard).canActivate(ctx: Context)  i tried but not working.

**Comment by serhiisol** [comment]
First of all, angular has its own decorator `@Injectable` and it has own container inside, and if you'll try to get that middleware from `@decorators/di` you'll definitely get an error. So you need to create anyway another middleware for express server or you can use express middleware inside of `CanActivate` guard of angular via getting it from `Container` from `@decorators/di` package.
